sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,804 @@
+"""
+This module contains Gradient Boosting Classifier and Regressor with
+Cross Validation.
+"""
+
+import warnings
+import numpy as np
+
+from ..base import BaseEstimator
+from ..metrics.scorer import check_scoring
+from ..utils.validation import check_X_y
+
+from .gradient_boosting import GradientBoostingClassifier
+from .gradient_boosting import GradientBoostingRegressor
+from ..tree._tree import DTYPE
+
+from ..model_selection import check_cv
+from ..model_selection import ParameterGrid
+from ..model_selection import ParameterSampler
+from ..model_selection._search import _CVScoreTuple
+
+from ..externals.joblib import Parallel, delayed
+
+
+class GradientBoostingClassifierCV(BaseEstimator):
+    """ Gradient Boosting Classification with Cross Validation
+
+    This class keeps adding estimators to the underlying GB estimator till
+    adding a new estimators does not improve the score by much. The goal of
+    this class is to determine the best choice of `n_estimators` for each set
+    of parameters. In most cases, this will be much faster than using
+    :class:`sklearn.model_selection.GridSearchCV` due to early stopping.
+    All parameters which are not parsed by this class are passed directly to
+    the underlying Gradient Boosting estimator. If one or more parameters is a
+    list, the Gradient Boosting model is evaluated for all elements of the
+    list.
+
+    Parameters
+    ----------
+    n_stop_rounds : int, optional, default=10
+        If the score on the test set rounded off to `score_precision` decimal
+        places does not change for `n_stop_rounds` iterations, the gradient
+        boosting is halted.
+
+    score_precision : int, optional, default=2
```

Comment:
Is this a good way to parametrize? It seems a bit odd to me. How does XGBoost do it? Maybe just "if it doesn't improve by ``epsilon`` within ``n_stop_rounds``? The way it is currently done is weirdly tied to the decimal system.